mesos-client/src/main/scala/mesosphere/mesos/client/MesosClient.scala

```diff
@@ -0,0 +1,514 @@
+    *
+    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#accept
+    */
+  def accept(accepts: Accept): Call = {
+    Call(
+      `type` = Some(Call.Type.ACCEPT),
+      accept = Some(accepts))
+  }
+
+  /** Sent by the scheduler to explicitly decline offer(s) received. Note that this is same as sending an ACCEPT
+    * call with no operations.
+    *
+    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#decline
+    */
+  def decline(offerIds: Seq[OfferID], filters: Option[Filters] = None): Call = {
+    Call(
+      `type` = Some(Call.Type.DECLINE),
+      decline = Some(Decline(offerIds = offerIds, filters = filters))
+    )
+  }
+
+  /** Sent by the scheduler to remove any/all filters that it has previously set via ACCEPT or DECLINE calls.
+    *
+    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#revive
+    */
+  def revive(roles: Option[String] = None): Call = {
+    Call(
+      `type` = Some(Call.Type.REVIVE),
+      revive = Some(Revive(role = roles))
+    )
+  }
+
+  /** Suppress offers for the specified roles. If `roles` is empty the `SUPPRESS` call will suppress offers for all
+    * of the roles the framework is currently subscribed to.
+    *
+    * http://mesos.apache.org/documentation/latest/upgrades/#1-2-x-revive-suppress
+    */
+  def suppress(roles: Option[String] = None): Call = {
+    Call(
+      `type` = Some(Call.Type.SUPPRESS),
+      suppress = Some(Call.Suppress(roles))
+    )
+  }
+
+  /** Sent by the scheduler to kill a specific task. If the scheduler has a custom executor, the kill is forwarded
+    * to the executor; it is up to the executor to kill the task and send a TASK_KILLED (or TASK_FAILED) update.
+    * If the task hasn’t yet been delivered to the executor when Mesos master or agent receives the kill request,
+    * a TASK_KILLED is generated and the task launch is not forwarded to the executor. Note that if the task belongs
+    * to a task group, killing of one task results in all tasks in the task group being killed. Mesos releases the
+    * resources for a task once it receives a terminal update for the task. If the task is unknown to the master,
+    * a TASK_LOST will be generated.
+    *
+    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#kill
+    */
+  def kill(taskId: TaskID, agentId: Option[AgentID] = None, killPolicy: Option[KillPolicy]): Call = {
+    Call(
+      `type` = Some(Call.Type.KILL),
+      kill = Some(Kill(taskId = taskId, agentId = agentId, killPolicy = killPolicy))
+    )
+  }
+
+  /** Sent by the scheduler to shutdown a specific custom executor. When an executor gets a shutdown event, it is
+    * expected to kill all its tasks (and send TASK_KILLED updates) and terminate.
+    *
+    * http://mesos.apache.org/documentation/latest/scheduler-http-api/#shutdown
+    */
+  def shutdown(executorId: ExecutorID, agentId: AgentID):Call = {
```

Comment:
Where does the ExecutorID name come from... Mesos? Can we make it ExecutorId? Preference towards consistence capitalization of Id, and, id is not an acronym.

Reply:
Yep, it's mesos type